core/components/header/header-section.tsx

```diff
@@ -0,0 +1,34 @@
+'use client';
+
+import { useParams } from 'next/navigation';
+import { ComponentProps, useCallback } from 'react';
+
+import { HeaderSection as VibesHeaderSection } from '@/vibes/soul/sections/header-section';
+import { usePathname, useRouter } from '~/i18n/routing';
+
+type Props = ComponentProps<typeof VibesHeaderSection>;
+
+const useSwitchLocale = () => {
+  const pathname = usePathname();
+  const router = useRouter();
+  const params = useParams();
```

Comment:
The `usePathname` and `useParams` hook require the component to be wrapped on a `Suspense` boundary otherwise the whole page will de-opt to client-side rendering.

In my work on PPR I'm having to rework components that use these hooks so that we can avoid this and have a proper fallback state. Introducing here at this level will be a bit problematic for those efforts 🙁